README.md

```diff
@@ -14,6 +14,22 @@ See [Plugins install instructions](https://theforeman.org/plugins/)
 for how to install Foreman plugins.
 You need to install the package `tfm-rubygem-foreman_userdata`.
 
+## Motivation
+
+This plug-in was developed to be used with VMWare vSphere provisioning from VMWare templates.
+vSphere offers the functionality to customize the templates during cloning. This is very handy for changing the network settings of a newly cloned VM.
+Foreman fully supports this feature via a userdata template. [The Foreman handbook](https://theforeman.org/manuals/1.17/index.html#image-provisioning-without-ssh) has details on how to configure this.
+An example template for a Linux host can be found in a [gist](https://gist.github.com/timogoebel/959deecff4f8ca8bcdd5fc9f95cbc8e3).
+The amount of things vSphere allows a user to customize are very limited, though. It's not possible to run a finish script or setup puppet with valid certificates on the new VM.
+This can be worked around by setting up a two step process: vSphere customization is just used to set up the network config of the host. Cloud-init is used to do the rest of the customization on first boot.
+
+This leads to this workflow:
+1. An administrator creates a vSphere VM template and sets up cloud-init as described below.
+2. Foreman creates a new cloned VM in vSphere and passes the [userdata template](https://gist.github.com/timogoebel/959deecff4f8ca8bcdd5fc9f95cbc8e3#file-userdata-template-erb) to vSphere.
+3. vSphere changes the network settings of the new host as defined in the userdata template and boots the new VM.
+4. The VM runs cloud-init and asks Foreman for the [cloud-init template](https://gist.github.com/timogoebel/959deecff4f8ca8bcdd5fc9f95cbc8e3#file-cloud-init-template-erb). Your VM needs network access to Foreman on ports `tcp/80` and `tcp/443`. Cloud-init then runs the template and executes the defined actions.
```

Comment:
@timogoebel , Thanks for updating this one, may be this is for RFE, but can client connect to proxy (capsule), on https (443) to get the template ? or it has to be master only and both 80 and 443 ?

Hope templates are not available at 8000 ?

Reply:
This would be a RFE, `foreman_userdata` does not have a smart proxy (capsule) plugin. So this is master only.